components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.hpp

```diff
@@ -303,10 +144,1035 @@ void mixing_ratio_to_cloud_mass(
+  const size_t my_size_ref = ncol * nlay * nlev;
+  pool_t::init(2e6 * (float(my_size_ref) / base_ref));
+
+  // We are now initialized!
+  initialized_k = true;
+}
+
+/*
+ * Compute band-by-band surface albedos from broadband albedos.
+ */
+template <typename SfcAlbDirVisT, typename SfcAlbDirNirT, typename SfcAlbDifVisT,
+          typename SfcAlbDifNirT, typename SfcAlbDirT, typename SfcAlbDifT>
+static void compute_band_by_band_surface_albedos(
+  const int ncol, const int nswbands,
+  SfcAlbDirVisT &sfc_alb_dir_vis, SfcAlbDirNirT &sfc_alb_dir_nir,
+  SfcAlbDifVisT &sfc_alb_dif_vis, SfcAlbDifNirT &sfc_alb_dif_nir,
+  SfcAlbDirT    &sfc_alb_dir,     SfcAlbDifT    &sfc_alb_dif)
+{
+  EKAT_ASSERT_MSG(initialized_k, "Error! rrtmgp_initialize must be called before GasOpticsRRTMGP object can be used.");
+  auto wavenumber_limits = k_dist_sw_k.get_band_lims_wavenumber();
+
+  EKAT_ASSERT_MSG(wavenumber_limits.extent(0) == 2,
+                  "Error! 1st dimension for wavenumber_limits should be 2. It's " << wavenumber_limits.extent(0));
+  EKAT_ASSERT_MSG(wavenumber_limits.extent(1) == static_cast<size_t>(nswbands),
+                  "Error! 2nd dimension for wavenumber_limits should be " + std::to_string(nswbands) + " (nswbands).");
+
+  // Loop over bands, and determine for each band whether it is broadly in the
+  // visible or infrared part of the spectrum (visible or "not visible")
+  Kokkos::parallel_for(MDRP::template get<2>({nswbands, ncol}), KOKKOS_LAMBDA(const int ibnd, const int icol) {
+
+    // Threshold between visible and infrared is 0.7 micron, or 14286 cm^-1.
+    const RealT visible_wavenumber_threshold = 14286;
+
+    // Wavenumber is in the visible if it is above the visible wavenumber
+    // threshold, and in the infrared if it is below the threshold
+    const bool is_visible_wave1 = (wavenumber_limits(0, ibnd) > visible_wavenumber_threshold ? true : false);
+    const bool is_visible_wave2 = (wavenumber_limits(1, ibnd) > visible_wavenumber_threshold ? true : false);
+
+    if (is_visible_wave1 && is_visible_wave2) {
+      // Entire band is in the visible
+      sfc_alb_dir(icol,ibnd) = sfc_alb_dir_vis(icol);
+      sfc_alb_dif(icol,ibnd) = sfc_alb_dif_vis(icol);
+    }
+    else if (!is_visible_wave1 && !is_visible_wave2) {
+      // Entire band is in the longwave (near-infrared)
+      sfc_alb_dir(icol,ibnd) = sfc_alb_dir_nir(icol);
+      sfc_alb_dif(icol,ibnd) = sfc_alb_dif_nir(icol);
+    }
+    else {
+      // Band straddles the visible to near-infrared transition, so we take
+      // the albedo to be the average of the visible and near-infrared
+      // broadband albedos
+      sfc_alb_dir(icol,ibnd) = 0.5*(sfc_alb_dir_vis(icol) + sfc_alb_dir_nir(icol));
+      sfc_alb_dif(icol,ibnd) = 0.5*(sfc_alb_dif_vis(icol) + sfc_alb_dif_nir(icol));
+    }
+  });
+}
+
+/*
+ * Compute broadband visible/UV and near-infrared surface fluxes.
+ */
+template <typename SwBndFluxDirT, typename SwBndFluxDifT, typename SfcFluxDirVisT,
```

Comment:
Same comment here as above, templating on each view seems overkill. It is perfectly reasonable to assume/require sw_bnd_flux_dir and sw_bnd_flux_dif to have the same type, sfc_flux_dir_vs, sfc_flux_dir_nir, sfc_flux_dif_vis, sfc_flux_dif_nir to have the same type.